onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -79,37 +79,38 @@ namespace Dml::GraphDescBuilder
         std::vector<DML_INTERMEDIATE_GRAPH_EDGE_DESC> graphIntermediateEdges;
         std::vector<DML_OUTPUT_GRAPH_EDGE_DESC> graphOutputEdges;
 
-        // Get the topological sorting of Lotus nodes
-        // paulm: breaking change from LOTUS that removed GetNodesInTopologicalOrder from Graph
-        onnxruntime::GraphViewer viewer(graph);
-        const std::vector<onnxruntime::NodeIndex>& orderedNodeIndices = viewer.GetNodesInTopologicalOrder();
-
         // Avoid using separate command lists for small graphs. This value can be reduced by tuning the 
         // flushing behavior of DmlCommandRecorder.  Its current behavior is to assume that graphs contain
         // enough GPU work to be worth flushing immediately.
         const uint32_t minNodeCountToReuseCommandList = 5;
         bool reuseCommandList = false;
         
-        if (orderedNodeIndices.size() >= minNodeCountToReuseCommandList)
+        if (indexedSubGraph.nodes.size() >= minNodeCountToReuseCommandList)
         {
             reuseCommandList = true;
         }
 
-        auto constantCpuGraphInputGetter = [&transferredInitializerMap](const std::string& argName)
+        auto constantCpuGraphInputGetter = [&isInitializerTransferable](const std::string& argName)
         {
             ComPtr<OnnxTensorWrapper> tensorWrapper;
 
-            auto iter = transferredInitializerMap.find(argName);
-            if (iter != transferredInitializerMap.end())
+            auto iter = isInitializerTransferable.find(argName);
+            if (iter != isInitializerTransferable.end())
             {
-                tensorWrapper = wil::MakeOrThrow<OnnxTensorWrapper>(&iter->second);
+                // Either the TensorProto is Input tensor or Output Tensor.
```

Comment:
This comment is confusing.   This is only used for inputs and the result is never modified.  The const_cast here is presumably just simpler than making surrounding code const correct